maintenance/wikia/AddDiscussionsToNavigation.php

```diff
@@ -0,0 +1,155 @@
+<?php
+
+/**
+ * Maintenance script to update navigation with Discussion links.
+ * @author Michal 'slayful' Turek
+ */
+
+ini_set( 'display_errors', 'stderr' );
+ini_set( 'error_reporting', E_NOTICE );
+
+require_once( dirname( __FILE__ ) . '/../Maintenance.php' );
+
+/**
+ * Class AddDiscussionsToNavigation
+ */
+class AddDiscussionsToNavigation extends Maintenance {
+
+	const VAR_NAME = 'wgOasisGlobalNavigation';
+	const NAVIGATION_ELEMENT = '**Special:DiscussionsNavigation|Discussions';
+	const FORUM_NAVIGATION_ELEMENT = '**Special:Forum|Forum';
+	const REASON = 'SOC-2816 add Discussion navigation to the navigation';
+	const SPLIT = "\r\n";
+
+	private $dryRun = false;
+	private $verbose = false;
+	private $rollback = false;
+
+	public function __construct() {
+		parent::__construct();
+		$this->mDescription =
+			"Update " . self::VAR_NAME . " with Discussions navigation " . self::NAVIGATION_ELEMENT;
+		$this->addOption( 'dry-run', 'Dry run mode', false, false, 'd' );
+		$this->addOption( 'verbose', 'Show extra debugging output', false, false, 'v' );
+		$this->addOption( 'rollback', 'Rollback the operation', false, false, 'r' );
+	}
+
+	public function execute() {
+		$this->dryRun = $this->hasOption( 'dry-run' );
+		$this->verbose = $this->hasOption( 'verbose' );
+		$this->rollback = $this->hasOption( 'rollback' );
+
+
+		$total = 0;
+		$successful = 0;
+		foreach ( $this->getAllSiteIds() as $siteId ) {
+			$success = $this->updateSiteId( trim( $siteId ) );
+			if ( $success ) {
+				$successful ++;
+			}
+			$total ++;
+		}
+
+		echo "\nTotal sites: $total, Success: {$successful}, Failed: " . ( $total - $successful ) .
+		     "\n\n";
+	}
+
+	private function getAllSiteIds() {
+		$id = WikiFactory::getVarIdByName( self::VAR_NAME );
+		$db = WikiFactory::db( DB_MASTER );
+		$sql =
+			( new WikiaSQL() )->SELECT( 'cv_city_id' )
+				->FROM( 'city_variables' )
+				->WHERE( 'cv_variable_id' )
+				->EQUAL_TO( $id );
+
+		return $sql->runLoop( $db, function ( &$siteIds, $row ) {
+			$siteIds[] = $row->cv_city_id;
+			$this->debug( "Added {$row->cv_city_id}" );
+		} );
+	}
+
+	private function updateSiteId( $siteId ) {
+		$varData = (array)WikiFactory::getVarByName( self::VAR_NAME, $siteId, true );
+		if ( empty( $varData['cv_id'] ) ) {
+			$this->debug( "Error:" . self::VAR_NAME . " not found.\n" );
+
+			return false;
+		}
+
+		if ( $varData['cv_variable_type'] !== 'text' ) {
+			echo $varData['cv_variable_type'];
+			die( "Error: " . self::VAR_NAME .
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use "static" keyword instead of "self". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS2037)